engine/src/main/java/org/terasology/reflection/metadata/ClassLibrary.java

```diff
@@ -60,6 +60,14 @@
      */
     <TYPE extends T> TYPE copy(TYPE object);
 
+    /**
+     * Copies the registered class, also making copies of any entities that it owns
+     *
+     * @param object The object to create a copy of
+     * @return A copy of the class, or null if not registered
+     */
+    <TYPE extends T> TYPE deepCopy(TYPE object);
```

Comment:
`recursivelyDeepCopy` ? 😁 

If we're sure we don't end up with multiple entity owners or circular ownership loops and silly things like that.

Reply:
That's probably a good idea. I'll change it.

I can think of one case where an ownership loop could potentially occur (assuming specific hypothetical module features that don't currently exist): if there's an item like the bags from some Minecraft mod I can't remember the name of, storage items that you can access directly from your inventory, it might be possible to put a bag in itself if that isn't specifically prevented. That would basically be a bug with this hypothetical module anyway though.